src/Helpers.js

```diff
@@ -0,0 +1,44 @@
+
```

Comment:
I'm not sure of the best way to approach this. I think it might be possible to use `JSON.parse()` with the optional second "reviver" argument. Then we could use a regex to find the original number in the JSON string and quote it. I think that would communicate the intent of the function more clearly. However I'm not sure if that would be a workable approach, so I'm very much open to discussion about it

Reply:
There's also this library: https://www.npmjs.com/package/json-bigint. Usually I'd be hesitant to add another dependency, but it seems it would only add a single transitive dependency

Reply:
It also has `options.storeAsString` to convert the bigint's to strings

Reply:
Afaik revivers work on parsed values. They'd allow pretty easily to quote numbers or convert to bigint numbers that are already in IEEE754 format

Reply:
Yes, it would be nice if we could use the reviver directly to modify the resulting JSON, but since the values would already be parsed, they would be JS `number`s, no? So we will have already lost precision by that point. My idea was to detect the numbers using the reviver (so we'd know we weren't in string, for example), but then modify the original JSON string

Reply:
I guess we could also just use a regex without using the reviver

Reply:
@ngua @klntsky thank you for being cautious on this.
Rory, I got your idea the second time I've read it, few hours later. I was thinking about both approaches you have mentioned before implementing low-level JS version. 

**1. Using revivers to detect numbers**
Your idea is to locate numbers -> replace them with a unique identifier -> use the identifiers locations to detect numbers locations in the string. There are some cons to this:
- efficiency
- we'd still need to parse the number manually
**2. Using regexes**
I have a feeling that the regexp correctly identifying (and matching) numbers would be at least as hard to write and read as the algorithm I have written. Do you think that it has more bugs? Maybe we could write a quickchen gen for particularly hard json strings and test it on that?

Reply:
Yes, that's basically what I meant (sorry if I didn't explain it very well). Good point about the efficiency aspect, so perhaps we should discard that option. For the regex, I'm not sure how bad it would be. Maybe one of us (or both) can write one and then we can decide on that option as well

I think the quickcheck idea is great, that would be a good guarantee regardless of what implementation we end up adopting

Reply:
You've spent more time than I have thinking about and solving this problem, so if you think that a low-level implementation is the most appropriate, I think that's what we should go with